.github/workflows/build-ros2-ws-action.yml

```
@@ -0,0 +1,91 @@
+name: CI
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: [master, feature/fix-github-workflows]
+  workflow_run:
+    workflows: [Build and Test Nightly]
+    types:  [requested, completed, in_progress]
+    branches: [feature/fix-github-workflows]
+jobs:
+  build-and-test:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
```

Please use the latest version (see [this](https://github.com/actions/checkout/releases)):
```suggestion
        uses: actions/checkout@v4.1.1
```